lib/Sema/CodeSynthesis.cpp

```diff
@@ -1346,8 +1346,9 @@ void swift::completeLazyVarImplementation(VarDecl *VD) {
   assert(!VD->isStatic() && "Static vars are already lazy on their own");
 
   // Create the storage property as an optional of VD's type.
-  SmallString<64> NameBuf = VD->getName().str();
-  NameBuf += ".storage";
+  SmallString<64> NameBuf;
+  NameBuf += "$__lazy_storage_";
```

Comment:
Nitpick: maybe separate the prefix from the name even harder? ObjC does `_$_` between sections, so `__lazy_storage_$_foo` or `$__lazy_storage_$_foo`. The former also makes it even less likely to conflict with a debugger variable, which probably only has the `$` at the beginning.

Reply:
I'm hesitant to write it without the `$` at the beginning, as that makes it a valid identifier. Do you think the potential name conflict matters there?

Reply:
Ah, I forgot we allowed `$` in non-initial positions. Okay.